### ISSUE TITLE
Error: Could not find or load main class

### ISSUE THREAD

**Issue by MarcinMoskala** [edvin/tornadofx-idea-plugin#47]
In following multiplatform project:
https://github.com/MarcinMoskala/KotlinAcademyApp

TornadoFx is running when started via gradle, but when I try to start it via TornadoFX plugin I get the following result: 
```
/usr/lib/jvm/java-8-oracle/bin/java -javaagent:/opt/idea/lib/idea_rt.jar=42641:/opt/idea/bin -Dfile.encoding=UTF-8 -classpath /usr/lib/jvm/java-8-oracle/jre/lib/charsets.jar:/usr/lib/jvm/java-8-oracle/jre/lib/deploy.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/jfxrt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/localedata.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-oracle/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-oracle/jre/lib/javaws.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jce.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfr.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jfxswt.jar:/usr/lib/jvm/java-8-oracle/jre/lib/jsse.jar:/usr/lib/jvm/java-8-oracle/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-oracle/jre/lib/plugin.jar:/usr/lib/jvm/java-8-oracle/jre/lib/resources.jar:/usr/lib/jvm/java-8-oracle/jre/lib/rt.jar com.marcinmoskala.kotlinacademy.desktop.MyApp

Error: Could not find or load main class com.marcinmoskala.kotlinacademy.desktop.MyApp
```

**Comment by edvin** [comment]
It sounds like the run configuration didn't select the `main` module of your Gradle project. Make sure it uses the module that contains the app class :) Does that help?

**Comment by MarcinMoskala** [comment]
I already have selected module with TornadoFX:
![run-debug configurations_121](https://user-images.githubusercontent.com/6850787/32992157-d6b9457c-cd47-11e7-8659-19802c206137.png)

**Comment by edvin** [comment]
Then it's obvious that the `desktop` module doesn't infact contain that class. Check the modules and see what they contain. I thought IDEA created a `_main` module with your app classes?

**Comment by MarcinMoskala** [comment]
I cannot select any other module because there is no Application found in any of other modules:
![run-debug configurations_125](https://user-images.githubusercontent.com/6850787/33011000-d605fb9e-cddc-11e7-8f66-89e470e5fee0.png)

Application is based in desktop module:
![selection_124](https://user-images.githubusercontent.com/6850787/33011021-e5e5a956-cddc-11e7-8e02-b44973752f57.png)

And it is in generated code:
![selection_123](https://user-images.githubusercontent.com/6850787/33011035-efba9130-cddc-11e7-929c-2e73d3a971c2.png)

I am not sure what do you meant by main. I haven't found Java entry point anywhere in the generated code. 

Note that the application is correctly running if you start it by:
```
gradle :desktop:run
```

**Comment by edvin** [comment]
IDEA by default creates three modules, even for a single module project. I think your gradle IDEA config is broken somehow. Try to reimport it. You can also try to create a new project based on the tornadofx-gradle-project template, and see how it sets up the modules. I tried now, created a project called `untitled12`, that resulted in the following three modules:

- untitled12
- untitled12_test
- untitled12_main

Only `untitled12_main` will allow me to run the app correctly.

The Gradle config for IDEA is completely confusing to me. Since it's working from the command line, it's obvious something is lacking in the IDEA config. I've seen so many bugs and issues with the Gradle IDEA support so I never use it myself, I'm afraid.

**Comment by liugangnhm** [comment]
> It sounds like the run configuration didn't select the `main` module of your Gradle project. Make sure it uses the module that contains the app class :) Does that help?

yes, the main module should be selected. this resolved my issue, thanks
